MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift

```diff
@@ -0,0 +1,156 @@
+extension MiniAppScriptMessageHandler {
+    func requestCustomPermissions(requestParam: RequestParameters?, callbackId: String) {
+        cachedUnknownCustomPermissionRequest.removeAll()
+        userRespondedRequestList.removeAll()
+        guard let requestParamValue = requestParam?.customPermissions else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppCustomPermissionError(customPermissionError: .invalidCustomPermissionRequest))
+            return
+        }
+
+        guard let miniAppPermissionRequestModelList = prepareCustomPermissionsRequestModelList(
+            permissionList: requestParamValue),
+            miniAppPermissionRequestModelList.count > 0 else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppCustomPermissionError(customPermissionError: .invalidCustomPermissionsList))
+            return
+        }
+        checkCustomPermissionsRequestStatusInCache(miniAppPermissionRequestModelList: miniAppPermissionRequestModelList, callbackId: callbackId)
+    }
+
+    func checkCustomPermissionsRequestStatusInCache(miniAppPermissionRequestModelList: [MASDKCustomPermissionModel], callbackId: String) {
+        /// storedPermissionsList - List of Permissions that is already stored in the cache
+        if let storedPermissionsList = self.miniAppStatus.getCustomPermissions(forMiniApp: self.miniAppId) {
+            /// filteredRequestModelList - List of Permissions that user hasn't responded yet.
+            let userNotRespondedRequestList = miniAppPermissionRequestModelList.filter {
+                !storedPermissionsList.contains($0)
+            }
+            userRespondedRequestList = storedPermissionsList.filter { miniAppPermissionRequestModelList.contains($0)
+            }
+            /// If the User hasn't responded to more than one permission that is requested, request Host app to request Custom Permissions
+            if userNotRespondedRequestList.count > 0 {
+                requestHostApp(customPermissionRequestList: userNotRespondedRequestList, callbackId: callbackId)
+            } else {
+                /// If the user has already responded to all the permissions that is requested filteredRequestModelList
+                /// will be 0 and the below code will filter the list of permissions(requested) from the stored permissions
+                /// and the same will be returned as cached response.
+                self.sendCachedSuccessResponse(result: userRespondedRequestList, callbackId: callbackId)
+            }
+        } else {
+            /// if User hasn't responded to any Custom permissions before
```

Comment:
Same here, I think this comment isn't needed. 

Also, see my comment above about filtering only `ALLOWED` permissions. I think that will simplify this logic because then we don't have a distinction between `Permissions which have been allowed`, `Permissions which have been denied`, and `Permissions which have not been requested`.

There should only be `Permissions which have been allowed` and `Permissions which are denied (need to be requested)`.

Reply:
> However, I think we should have only the concept of "Accepted" and "Denied". So if a user chooses "Deny" for a certain permission, they can still be prompted about that permission in the future if it's requested again. This will make some of the things in the SDK simpler, it allows the Host App and Mini App to have more flexibility (the Mini App can still request a Denied permission again), and it allows us to provide all of the supported permissions when returning `getCustomPermissions`, rather than just those which have been requested before.

Thanks for the comment cory. Yes, I implemented something like iOS behavior, where permission which is denied by the user cannot be requested again. I will make changes to as per the comments and to retrieve supported permissions when `getCustomPermissions`  is called